### DISCUSSION TITLE
Loading state in JavaScript function

### DISCUSSION THREAD

**Discussion by ppegu** [livewire/livewire#1754] [4. General]
How i get if an action  is running and we are waiting for network request in JavaScript function ? like Livewire.onLoading((event) => { console.log('loading') } );    .. ??

**Comment by Rishab49** [comment]
You can use Javascript Promises for this with async function
check out : https://javascript.info/promise-basics

**Comment by Rishab49** [comment]
This block of code can give better idea
> var res;
        async function main(){
            var promise=new Promise((resolve,reject)=>{
                setTimeout(function(){
                    resolve('starting');
                    console.log('first step complete');
                },2000);
            });
            res=await promise;
            return res;
         }
       main();
      var mainInterval=setInterval(function(){
           if(res==undefined){
               console.log('loading..');
           }
           if(res=="starting"){
               console.log('end');
               clearInterval(mainInterval);
           }
       },100);

**Reply by ppegu** [reply]
But why would be do in pure js blocks instead using livewire cool things... I meant to say .. in livewire there's a function for disabled buttons on click as `wire:loading.attr` and I'm directly asking is can we do run a function while it's loadong at the same time `wire:loading` occurs ends on its end

**Comment by calebporzio** [comment]
Yeah, I've thought it would be cool to have something like `$wire.loading(() => {})` or something myself.
